### ISSUE TITLE
Calibrating the confidence level

### ISSUE THREAD

**Issue by Carnaux** [Carnaux/NFT-Marker-Creator#31]
@kalwalt @nicolocarpignoli @jonasjohansson With the last commit I added a confidence level based on Shannon's entropy equation. To calculate the level( 0 - 5 ) I tested with [this image](https://www.google.com/url?sa=i&source=images&cd=&ved=2ahUKEwjmkcj61PzmAhUVHbkGHU1wBykQjRx6BAgBEAQ&url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FMinimalist_photography&psig=AOvVaw2gVGuTjKqtWFJZ2ZS7LpVH&ust=1578870456899571) that gives an entropy of 4.6, I couldn't get it to be detected by my webcam so i set its entropy as the minimum (0), the maximum(5) its an entropy of 5.17 of the pinball image. 

The node app logs at the end: 
`
Confidence level: [ * * * * * ] 5/5 || Entropy: 5.17 || Current max: 5.17 min: 4.6
`
And the web version display stars.

If you find any image that does not correspont to it's confidence level, please report in this issue!

**Comment by kalwalt** [comment]
Interesting feature! IT would be interesting made some examples as user cases to show which images are better and which is not.

**Comment by Carnaux** [comment]
I'm writing this [wiki page](https://github.com/Carnaux/NFT-Marker-Creator/wiki/Creating-good-markers) to guide the users.

**Comment by kalwalt** [comment]
Perfect!

**Comment by pepperface** [comment]
Awesome! Huge thanks, @Carnaux :+1:

**Comment by nicolocarpignoli** [comment]
> I'm writing this [wiki page](https://github.com/Carnaux/NFT-Marker-Creator/wiki/Creating-good-markers) to guide the users.

this issue can be useful for the wiki: https://github.com/Carnaux/NFT-Marker-Creator/issues/11

after you put those info on the wiki you also may close the issue :)

**Comment by Carnaux** [comment]
I've just finished the wiki! What do you think? @kalwalt @nicolocarpignoli

**Comment by kalwalt** [comment]
That Is very good @Carnaux ! I will make a link as i said in the jsartoolkit5 README.md

**Comment by nicolocarpignoli** [comment]
for me it's super ok!

Just two point:

- maybe there is in general 'too much caution' when talking about "good markers". What I mean is that while reading the Wiki, a common user may think that to create a good marker is hard and complicated, while in general most of the required variables are pretty much satisfied every time.

- we can also add a quick mention to the node generator and web generator: to link them in the 'home' section will be enough.

If you'd like I can make the modifies but I don't know how to do that with the wiki.

**Comment by kalwalt** [comment]
@nicolocarpignoli to modify the wiky Just fork It as another repository 😃

**Comment by nicolocarpignoli** [comment]
yes I know but the proposed methods are not very straightforward to me: https://stackoverflow.com/questions/10642928/how-to-pull-request-a-wiki-page-on-github

I can propose some change in PVT if @Carnaux has no time for that :) Let  me know

**Comment by kalwalt** [comment]
Sorry, didn't know of PR complexity for the wiki, probably better suggest changes or send a .zip file.

**Comment by Carnaux** [comment]
I will write other pages on the wiki later, and I will make the modifications

**Comment by Carnaux** [comment]
I just added a little bit of text that may help with: 

> a common user may think that to create a good marker is hard and complicated

**Comment by nicolocarpignoli** [comment]
Hi there! 
I have problem in understanding the confidence level.
Two examples:

this image: https://github.com/nicolocarpignoli/nft-images/blob/master/tardis.jpg
IS recognized, and gives this confidence level:
` [----- ] 0.96/5 || Entropy: 4.71 || Current max: 5.17 min: 4.6`

this other image: https://github.com/nicolocarpignoli/nft-images/blob/master/room104.jpg
IS NOT recognized, and gives this confidence level:
`Confidence level: [ * * * *- ] 4.3/5 || Entropy: 5.09 || Current max: 5.17 min: 4.6`

I'm a bit  confused @Carnaux

**Comment by kalwalt** [comment]
> Hi there!
> I have problem in understanding the confidence level.
> Two examples:
> 
> this image: https://github.com/nicolocarpignoli/nft-images/blob/master/tardis.jpg
> IS recognized, and gives this confidence level:
> ` [----- ] 0.96/5 || Entropy: 4.71 || Current max: 5.17 min: 4.6`
> 
> this other image: https://github.com/nicolocarpignoli/nft-images/blob/master/room104.jpg
> IS NOT recognized, and gives this confidence level:
> `Confidence level: [ * * * *- ] 4.3/5 || Entropy: 5.09 || Current max: 5.17 min: 4.6`
> 
> I'm a bit confused @Carnaux

I think that images are ok but maybe the problem is the resolution: 
tardis.jpg is  627 kb room104.jpg is 269 kb you should look also the number of imageSet and number of FeaturesPoints and compare both.

**Comment by Carnaux** [comment]
Strange, I will test it and give you feedback.

**Comment by Carnaux** [comment]
@nicolocarpignoli I tested both images, and both tracked. The Tardis image got a low score because of the color palette.

**Comment by nicolocarpignoli** [comment]
Yes it was an error of mine (not regardin marker generator). Marker generator works great. I will only make more clearer the confidence score maybe?

**Comment by nicolocarpignoli** [comment]
Maybe, if confidence is under a certain treshold (2 points?) we can link on the confidence output the URL of the doc on how to choose better images.

**Comment by Carnaux** [comment]
I will search for a quick and costless feature detection algorithm, that way it will be more precise

**Comment by patrickoshaughnessey** [comment]
@Carnaux did you ever find anything that is a more reliable measurement of the confidence? We are also having an issue with this.

**Comment by mchangxe** [comment]
Hi guys, thanks for adding in this feature, but I don't think its working as intended. Other than the pinball picture, any picture i upload onto the web version of the tool gives back 0/5 confidence (5 stars all greyed out). Any ideas?

**Comment by Carnaux** [comment]
Hey guys, I'm totally out of time, so I can't solve it now, when I get some free time I can try to solve it
